query/file_log_watcher.go

```diff
@@ -0,0 +1,178 @@
+package query
+
+import (
+	"errors"
+	"os"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	l "github.com/influxdata/influxdb/logger"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type FileLogWatcher struct {
+	path            string
+	currFile        *os.File
+	logger          *zap.Logger
+	formatterConfig string
+	executor        *Executor
+	mu              sync.Mutex
+	fsWatcher       *fsnotify.Watcher
+}
+
+func (f *FileLogWatcher) Event() <-chan fsnotify.Event {
+	return f.fsWatcher.Events
+}
+
+func (f *FileLogWatcher) Error() <-chan error {
+	return f.fsWatcher.Errors
+}
+
+func (f *FileLogWatcher) Add(name string) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	return f.fsWatcher.Add(name)
+}
+
+func (f *FileLogWatcher) Remove(name string) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	return f.fsWatcher.Remove(name)
+}
+
+func NewFileLogWatcher(e *Executor, path string, logger *zap.Logger, format string) *FileLogWatcher {
+	// FileWatcher is not meant to be used with interactive shell sessions so
+	// console as a logging format will not be supported.
+	if format == "console" {
+		logger.Error("unknown logging format", zap.String("format", format), zap.String("path", path))
```

Comment:
I feel like it would be cleaner for `NewFileLogWatcher` to return an additional error value and not take the logger object in. The caller should log the error returned. 

Stylistically, a function that on error only returns `nil` feels like it should be left in the world of C.